PURL-SPECIFICATION.rst

```diff
@@ -257,31 +257,26 @@ These ``purl`` separator characters MUST NOT be percent-encoded when used as
 Percent-encoding rules
 ----------------------
 
-Unless otherwise provided in this specification, when applying percent-encoding
-or decoding to a string, use the rules of RFC 3986 section 2
-(https://datatracker.ietf.org/doc/html/rfc3986#section-2).  In the event of any
-conflict between this specification and RFC 3986 section 2, this specification
-governs.
-
-In the "Rules for each ``purl`` component" section above, each component
-defines when and how to apply percent-encoding and decoding to its content.
-
-When percent-encoding is required, all Permitted Characters MUST be encoded as
-UTF-8 and then percent-encoded except for the following:
-
-- the alphanumeric characters,
-
-- the ASCII characters ``.-_~`` (period '.', dash '-', underscore
-  '_' and tilde '~'),
-
-- the percent sign '%' when used to represent a percent-encoded character,
-
-- a ``purl`` separator when being used as a ``purl`` separator, and
-
-- the colon ':', whether used as a ``purl`` separator or otherwise.
-
-In addition, where the space ' ' is permitted, it MUST be percent-encoded as
-'%20'.
+- In the "Rules for each ``purl`` component" section above, each component
+  defines when and how to apply percent-encoding and decoding to its content,
+  including which characters to percent-encode and when percent-encoding is
+  required.
+- When percent-encoding is required by a component definition, each
```

Comment:
For simplicity's sake I would say that percent encoding is always required. If a component only allows characters from the unreserved set, then percent-encoding simply becomes the identity function.

Reply:
Thanks @ppkarwasz .  I have seen that point asserted elsewhere (that percent-encoding is always required) but that's contrary to my understanding that percent-encoding is only required under certain defined circumstances (both in RFC 3986 and in the PURL spec).  In the PURL spec, it's the component definitions that instruct WHICH characters and WHEN, and the `type` definition, for example, makes clear that `The type MUST NOT be percent-encoded.`

Reply:
There are two concepts of "percent-encoding": the [percent encoding of a byte](https://url.spec.whatwg.org/#percent-encode) and the [percent encoding after encoding of a string](https://url.spec.whatwg.org/#string-percent-encode-after-encoding).
Some components like `type` will never need to use the former, but they can certainly used the latter. The "percent encoding after encoding" of `maven` will be simply `maven`.

Reply:
@ppkarwasz let's refine the wording alright. But we use RFC 3986 encoding rules for each "octet" and we define which character needs this treatment and when. Blanket encoding everything will be a source of confusion IMHO.